crates/runtime/src/mmap.rs

```diff
@@ -412,11 +416,24 @@ impl Mmap {
             "changing of protections isn't page-aligned",
         );
 
-        region::protect(
-            self.as_ptr().add(range.start),
-            range.end - range.start,
-            region::Protection::READ_EXECUTE,
-        )?;
+        let address = self.as_ptr().add(range.start);
+        let size = range.end - range.start;
+
+        if enable_branch_protection {
+            #[cfg(all(target_arch = "aarch64", target_os = "linux"))]
+            if std::arch::is_aarch64_feature_detected!("bti") {
```

Comment:
Could this perhaps be unconditionally done? I would expect that the detection of the `bti` feature would happen at a higher level than this (e.g. at the Wasmtime `Config` layer) where `enable_branch_protection` is only `true` if the machine actually has `bti`.

Although I think it's ok to still have `cfg!(target_arch = "aarch64")` in the condition since this is still arm64-specific.

Reply:
> Could this perhaps be unconditionally done? I would expect that the detection of the `bti` feature would happen at a higher level than this (e.g. at the Wasmtime `Config` layer) where `enable_branch_protection` is only `true` if the machine actually has `bti`.

My other comment provides a bit of context, but I didn't touch the higher-level layers on purpose, so that we support as much flexibility as possible, e.g. compile ahead-of-time a module with BTI on a machine that does not support BTI, or load a module that has been compiled ahead-of-time without BTI, while the engine is set to use BTI. I think that we will need to be more careful if we are making these decisions at a higher level - for instance, distinguish between `wasmtime compile` and `wasmtime run`.